### ISSUE TITLE
ReplayDataLoader data-memory ratio

### ISSUE THREAD

**Issue by HamedHemati** [ContinualAI/avalanche#833]
The current implementation of `ReplayPlugin` uses `ReplayDataLoader` before each training experience to augment the main data loader with the samples from its buffer. By inspecting the task labels in a task-incremental setup I noticed that `ReplayDataLoader` always tries to divide each mini-batch equally among all seen tasks. There seems to be no option for a fixed data-buffer ratio. There is just one parameter `ratio_data_mem` in the documentation that is not actually used in the implementation. 

Two potential problems with the current implementation that I can think of:
- Overfitting to previous tasks as the ratio of samples from the current task decreases over time: only 1/N of samples will be from the new task (N: number of observed tasks)
- Some algorithms use a 50-50 loading mechanism: 50% of each batch is from the new task data and 50% comes the buffer samples 

Is there an option that I'm missing in `ReplayDataLoader`?

If not, then the question would be: should we extend the current `ReplayDataLoader` or should we create another data loader for cases where the ratio of data-buffer is fixed?

**Comment by AntonioCarta** [comment]
You are correct. We should fix it by adding (and using) `ratio_data_mem`. Notice that other dataloaders may already do what you need (e.g. `GroupBalancedDataLoader`).

@hamedhemati do you want to do this? I think it's an easy change.

**Comment by hamedhemati** [comment]
@AntonioCarta actually the `force_data_batch_size` parameter in `ReplayDataLoader` can also do the same thing (by knowing the train batch size). There was just a minor mistake in mem_batch_size that I've fixed.  I thought maybe it's just simpler to add the `force_data_batch_size` option directly to ReplayPlugin. This way the user can directly set the number of samples dedicated to the data part when creating a replay plugin.

I've made a PR to add this option. I've tested it locally with a task-incremental scenario and the behavior is as expected.

There is also another point regarding how the data is loaded for the memory part. Forcing the memory data loader to be task-balanced can also reduce the flexibility replay mechanism. This can also raise an error when `mem_batch_size` < # of observed tasks; for example when `train_batch_size=32` and you use 20 samples for the data part,   then you have only 12 slots left for the memory part, which means the data loader creation step will crash after the 12th task.

Should I create a separate issue for this or do we just continue the discussion here?
